### PR TITLE
[SofaConstraint] ADD control on constraint force in UniformConstraint

### PR DESCRIPTION
**Pull request by adagolodjo**
Force the constraint force to zero or restPosition

As @damienmarchal said, this is a new version of what Christian proposed in PR #1022 the only difference is that Damien propose to change all the for loops (inside the getConstraintViolation function) by one lambda function. And I will complete this PR next weeks by doing this in all constraints component.   

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: modules/SofaConstraint/UniformConstraint.inl [hunk 1]
```diff
@@ -13,8 +13,8 @@ namespace constraint
 
 template< class DataTypes >
 UniformConstraint<DataTypes>::UniformConstraint()
-    :d_iterative(initData(&d_iterative, true, "iterative", "Iterate over the bilateral constraints, otherwise a block factorisation\
-                                                            is computed."))
+    :d_iterative(initData(&d_iterative, true, "iterative", "Iterate over the bilateral constraints, otherwise a block factorisation is computed."))
+    ,d_constraintRestPos(initData(&d_constraintRestPos, false, "constraintRestPos", "if false, constraint the pos to be zero / if true constraint the current position to stay at rest position"))
```
Comment (damienmarchal) [inline modules/SofaConstraint/UniformConstraint.inl]: a verb... constrainToRestPos

File: modules/SofaConstraint/UniformConstraint.inl [hunk 2]
```diff
@@ -51,38 +64,34 @@ void UniformConstraint<DataTypes>::getConstraintViolation(const sofa::core::Cons
...
-        const std::size_t N = Deriv::size();
-
-        for (std::size_t i = 0; i < vfree.size(); ++i)
-        {
-            for (std::size_t j = 0; j < N; ++j)
-            {
-                resV->set(m_constraintIndex + i*N + j, vfree[i][j] + invDt * pos[i][j] );
-            }
+        if (d_constraintRestPos.getValue()){
+            computeViolation(resV, m_constraintIndex, vfree, Deriv::size(),[&invDt,&pos,&vfree,&restPos](int i, int j)
```
Comment (damienmarchal) [inline modules/SofaConstraint/UniformConstraint.inl:73]: lambda :+1:

**Comment by damienmarchal** [comment]
Hello Youness,

Thank for the PR, could you explain abit if this is a new version of the one made by @ChristianDuriez (#1022)

File: modules/SofaConstraint/UniformConstraint.inl (hunk 2 above)
Comment (guparan) [reply modules/SofaConstraint/UniformConstraint.inl:73]: Lambda is indeed nice but the indentation is strange.
I'd suggest 
```cpp
computeViolation( resV, m_constraintIndex, vfree, Deriv::size(), [&invDt, &pos, &vfree, &restPos](int i, int j) {
    return vfree[i][j] + invDt * ( pos[i][j] - restPos[i][j] );
});
```

**Comment by damienmarchal** [comment]
[ci-build][with-scene-tests][with-regression-tests]

**Pull request merged**